crates/sui-node/src/lib.rs

```diff
@@ -371,26 +379,127 @@ impl SuiNode {
 
-        let (tx_reconfigure_consensus, rx_reconfigure_consensus) = channel(100);
+        let consensus_adapter =
+            Self::construct_consensus_adapter(consensus_config, state.clone(), prometheus_registry);
 
-        let validator_service = ValidatorService::new(
+        let validator_server_handle = Self::start_grpc_validator_service(
             config,
             state.clone(),
-            checkpoint_store,
-            state_sync_handle.clone(),
+            consensus_adapter.clone(),
+            &prometheus_registry,
+        )
+        .await?;
+
+        let narwhal_manager = Self::construct_and_run_narwhal_manager(
+            config,
+            consensus_config,
+            consensus_adapter.clone(),
+            checkpoint_store.clone(),
+            state.clone(),
+            state_sync_handle,
+            &prometheus_registry,
+        )
+        .await?;
+
+        Ok((validator_server_handle, narwhal_manager))
+    }
+
+    async fn construct_and_run_narwhal_manager(
+        config: &NodeConfig,
+        consensus_config: &ConsensusConfig,
+        consensus_adapter: Arc<ConsensusAdapter>,
+        checkpoint_store: Arc<CheckpointStore>,
+        state: Arc<AuthorityState>,
+        state_sync_handle: state_sync::Handle,
+        prometheus_registry: &Registry,
+    ) -> Result<NarwhalManager> {
+        let checkpoint_output = Box::new(SubmitCheckpointToConsensus {
+            sender: consensus_adapter.clone(),
+            signer: state.secret.clone(),
+            authority: config.protocol_public_key(),
+        });
+
+        let certified_checkpoint_output = SendCheckpointToStateSync::new(state_sync_handle);
+
+        let checkpoint_service = CheckpointService::spawn(
+            state.clone(),
+            checkpoint_store.clone(),
+            Box::new(state.database.clone()),
+            checkpoint_output,
+            Box::new(certified_checkpoint_output),
+            CheckpointMetrics::new(prometheus_registry),
+        );
+        let committee = config.genesis()?.narwhal_committee().load();
+
+        let narwhal_config = NarwhalConfiguration {
+            primary_keypair: config.protocol_key_pair().copy(),
+            network_keypair: config.network_key_pair.copy(),
+            worker_ids_and_keypairs: vec![(0, config.worker_key_pair().copy())],
+            worker_cache: config.narwhal_worker_cache()?,
+            storage_base_path: consensus_config.db_path().to_path_buf(),
+            parameters: consensus_config.narwhal_config().to_owned(),
+            execution_state: Arc::new(ConsensusHandler::new(state.clone(), checkpoint_service)),
+            tx_validator: SuiTxValidator::new(state.clone(), &prometheus_registry),
+        };
+
+        let (tx_start, tr_start) = channel(1);
+        let (tx_stop, tr_stop) = channel(1);
+        let join_handle =
+            spawn_monitored_task!(run_narwhal_manager(narwhal_config, tr_start, tr_stop));
```

Comment:
You can now use the `spawn_logged_monitored_task` that has been introduced here https://github.com/MystenLabs/sui/pull/6696 in order to log when the task has finished/shutdown